File: kotlin/vci-client/src/main/java/io/mosip/vciclient/dto/IssuerMetaData.kt
package io.mosip.vciclient.dto

import io.mosip.vciclient.constants.CredentialFormat

@Deprecated(
    message = "IssuerMetaData is deprecated. Use IssuerMetadata instead as per the new VCIClient library contract.",
    replaceWith = ReplaceWith("IssuerMetadata"),
    level = DeprecationLevel.WARNING
)
@Suppress("ArrayInDataClass")
data class IssuerMetaData(
    val credentialAudience: String,
    val credentialEndpoint: String,
    val downloadTimeoutInMilliSeconds: Int,
    val credentialType: Array<String>? = null,
    val credentialFormat: CredentialFormat,
    val doctype: String? = null,
    val claims: Map<String, Any>? = null,
)

File: js/src/dto/issuerMetaData.js
class IssuerMetaData {
    #credentialAudience;
    #credentialEndpoint;
    #downloadTimeoutInMillSeconds;
    #credentialType;
    #credentialFormat;

    /**
     * @param {string} credentialAudience
     * @param {string} credentialEndpoint
     * @param {number} downloadTimeoutInMillSeconds
     * @param {[string]} credentialType
     * @param {string} credentialFormat
     */
    constructor(credentialAudience, credentialEndpoint, downloadTimeoutInMillSeconds, credentialType, credentialFormat) {
        this.#credentialAudience = credentialAudience;
        this.#credentialEndpoint = credentialEndpoint;
        this.#downloadTimeoutInMillSeconds = downloadTimeoutInMillSeconds;
        this.#credentialType = credentialType;
        this.#credentialFormat = credentialFormat;
    }

    get credentialAudience() {
        return this.#credentialAudience;
    }

    get credentialEndpoint() {
        return this.#credentialEndpoint;
    }

    get downloadTimeoutInMillSeconds() {
        return this.#downloadTimeoutInMillSeconds;
    }

    get credentialType() {
        return this.#credentialType;
    }

    get credentialFormat() {
        return this.#credentialFormat;
    }
}

module.exports = IssuerMetaData;